lib/cliapp/src/Obelisk/CliApp/Process.hs

```diff
@@ -242,6 +252,12 @@ streamToLog stream = fix $ \loop -> do
 waitForProcess :: MonadIO m => ProcessHandle -> m ExitCode
 waitForProcess = liftIO . Process.waitForProcess
 
+-- | Converts an 'ExitCode' to an exception when it's non-zero.
+exitCodeToException :: (CliThrow e m, AsProcessFailure e) => ProcessSpec -> ExitCode -> m ()
+exitCodeToException spec = \case
+  ExitSuccess -> pure ()
+  ExitFailure code -> throwError $ review asProcessFailure $ ProcessFailure (Process.cmdspec $ _processSpec_createProcess spec) code
```

Comment:
Does this resolve the `TODO` on line 233 ?

Reply:
No this just turns `ExitCode` into exception but any actual exceptions pass through as normal. Now that I think about it that TODO is probably just wrong and I should delete it.